doc/source/whatsnew/v0.17.0.txt

```diff
@@ -80,6 +80,8 @@ Other enhancements
 
 - ``regex`` argument to ``DataFrame.filter`` now handles numeric column names instead of raising ``ValueError`` (:issue:`10384`).
 
+- ``merge`` will now allow duplicate column names if they are not merged upon (:issue:`10639`).
+
```

Comment:
i don't think there is a need for an example since the behavior now is very natural. Merging 'just works' despite some duplicate column names which are irrelevant to the merge process.